server/server.go

```diff
@@ -259,27 +262,15 @@ func (server *Server) InitSubscription() {
 }
 
 // Run will run the server.
-func (server *Server) Run() error {
+func (server *Server) Run(ctx context.Context) error {
 	if !server.readonly {
-		if err := server.TaskScheduler.Run(); err != nil {
-			return err
-		}
-
-		if err := server.TaskCheckScheduler.Run(); err != nil {
-			return err
-		}
-
-		if err := server.SchemaSyncer.Run(); err != nil {
-			return err
-		}
-
-		if err := server.BackupRunner.Run(); err != nil {
-			return err
-		}
-
-		if err := server.AnomalyScanner.Run(); err != nil {
-			return err
-		}
+		go server.TaskScheduler.Run(ctx, &server.runnerWG)
+		go server.TaskCheckScheduler.Run(ctx, &server.runnerWG)
+		go server.SchemaSyncer.Run(ctx, &server.runnerWG)
+		go server.BackupRunner.Run(ctx, &server.runnerWG)
+		go server.AnomalyScanner.Run(ctx, &server.runnerWG)
+		// Add above 5 goroutines to the wait group.
+		server.runnerWG.Add(5)
```

Comment:
I am inclined to repeat 5 times for server.runnerWG.Add.

Otherwise we may forget to change the amount when adding a new runner

Reply:
+1 Done.